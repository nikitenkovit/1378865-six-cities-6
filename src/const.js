export const OfferType = {
  APARTMENT: `Apartment`,
  ROOM: `Private Room`,
  HOUSE: `House`,
  HOTEL: `Hotel`
};

export const PathValue = {
  MAIN_SCREEN: `/`,
  FAVORITES_SCREEN: `/favorites`,
  ROOM_SCREEN: `/offer/:id`
};

export const CityCoordinate = {
  AMSTERDAM: {
    latitude: 52.38333,
    longitude: 4.9,
    zoom: 12
  }
}; // скорее всего временно

export const OffersListClassName = {
  CITY_PLACES: `cities__places-list`,
  NEAR_PLACES: `near-places__list`
};

export const BookmarkButtonProperty = {
  ROOM_SCREEN: {
    buttonClass: `property__bookmark-button`,
    iconClass: `property__bookmark-icon`,
    iconWidth: `31`,
    iconHeight: `33`
  },
  OFFER_CARD: {
    buttonClass: `place-card__bookmark-button`,
    iconClass: `place-card__bookmark-icon`,
    iconWidth: `18`,
    iconHeight: `19`
  }
};
