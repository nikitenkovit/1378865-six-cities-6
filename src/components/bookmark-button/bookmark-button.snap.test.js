import React from "react";
import * as redux from 'react-redux';
import {Router} from 'react-router-dom';
import {render} from "@testing-library/react";
import configureStore from 'redux-mock-store';
import BookmarkButton from "./bookmark-button";
import {BookmarkButtonProperty} from "../../const";
import browserHistory from "../../history";

const mock = {
  id: 1,
  isFavorite: false,
  bookmarkButtonProperty: BookmarkButtonProperty.OFFER_CARD
};

const mockStore = configureStore({});

it(`Should BookmarkButton render correctly`, () => {
  const {container} = render(
      <redux.Provider store={mockStore({})}>
        <Router history={browserHistory}>
          <BookmarkButton
            id={mock.id}
            isFavorite={mock.isFavorite}
            bookmarkButtonProperty={mock.bookmarkButtonProperty}
          />
        </Router>
      </redux.Provider>);
  expect(container).toMatchSnapshot();
});
