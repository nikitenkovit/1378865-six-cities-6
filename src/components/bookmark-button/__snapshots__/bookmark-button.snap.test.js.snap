// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should BookmarkButton render correctly 1`] = `
<div>
  <button
    class="place-card__bookmark-button button"
    type="button"
  >
    <svg
      class="place-card__bookmark-icon"
      height="19"
      width="18"
    >
      <use
        xlink:href="#icon-bookmark"
      />
    </svg>
    <span
      class="visually-hidden"
    >
      To bookmarks
    </span>
  </button>
</div>
`;
